trace-sdk/src/main/java/io/bitrise/trace/network/adapters/PointAdapter.java

```diff
@@ -37,29 +37,31 @@ public Point deserialize(@NonNull final JsonElement json, @NonNull final Type ty
     if (srcValueCase != null) {
       final Point.ValueCase valueCase = Point.ValueCase.forNumber(srcValueCase.getAsInt());
 
-      switch (valueCase) {
-        case INT64_VALUE:
-          pointBuilder.setInt64Value(srcPoint.get(PROPERTY_VALUE).getAsInt());
-          break;
-        case DOUBLE_VALUE:
-          pointBuilder.setDoubleValue(srcPoint.get(PROPERTY_VALUE).getAsDouble());
-          break;
-        case DISTRIBUTION_VALUE:
-          final DistributionValue distributionValue = NetworkClient.getGson().fromJson(
-              srcPoint.get(PROPERTY_VALUE), DistributionValue.class);
-          pointBuilder.setDistributionValue(distributionValue);
-          break;
-        case SUMMARY_VALUE:
-          final SummaryValue summaryValue = NetworkClient.getGson().fromJson(
-              srcPoint.get(PROPERTY_VALUE), SummaryValue.class);
-          pointBuilder.setSummaryValue(summaryValue);
-          break;
-        case VALUE_NOT_SET:
-          // we deliberately do not deserialize values that were not set.
-          break;
-        default:
-          // nop.
-          break;
+      if (valueCase != null) {
```

Comment:
this was a warning in android studio - and i accidentally wrote a test that caught it. My assumption was `Point.ValueCase.forNumber(srcValueCase.getAsInt());` would return the number even if it couldn't find it, and then it would hit the default block in the switch - however it looks like the method will return a null instead - so this needs to be checked it's not null.